src/PETScNonlinearSolvers.jl

```diff
@@ -71,27 +73,6 @@ function snes_jacobian(csnes:: Ptr{Cvoid},
   return PetscInt(0)
 end
 
```

Comment:
I have realised that we are allocating extra vectors in `snes_residual` and `snes_jacobian` under-the-hood.

`residual!` and `jacobian!` expect vectors aligned with the FESpace. If you pass vectors aligned with the system, then a temporary copy will be created at https://github.com/gridap/GridapDistributed.jl/blob/19d25a015074a1770c7fb602a413bfaeee1fdc7f/src/Algebra.jl#L64

Solution. Store `x` in the cache (i.e., the vector passed to `solve!`) and then we can do something like:

```julia
copy!(cache.x_julia_vec, Vec(cx))
copy!(cache.x,cache.x_julia_vec)
  jacobian!(cache.jac_julia_mat_A,cache.op,cache.x)
```
Idem for the residual.